### PR TITLE
scatter if operation

### PR DESCRIPTION
**Pull request by jpola**
Hi,

In my project I needed the scatter if operation so I thought I will also push it here. Could you please check it if this is useful?

Thanks,

### CODE REVIEW

File: include/boost/compute/algorithm/scatter_if.hpp [hunk 1]
```diff
@@ -0,0 +1,119 @@
+//---------------------------------------------------------------------------//
+// Copyright (c) 2013 Kyle Lutz <kyle.r.lutz@gmail.com>
```
Comment (haahh) [inline include/boost/compute/algorithm/scatter_if.hpp]: You should put your name here ;)

File: test/test_scatter_if.cpp [hunk 2]
```diff
@@ -0,0 +1,141 @@
+//---------------------------------------------------------------------------//
+// Copyright (c) 2013 Kyle Lutz <kyle.r.lutz@gmail.com>
```
Comment (haahh) [inline test/test_scatter_if.cpp]: and here!

**Comment by kylelutz** [comment]
Looks great, thanks for adding this!

One issue with initialization in the tests (from the clang build in travis-ci):
```
[ 42%] Building CXX object test/CMakeFiles/test_search.dir/test_search.cpp.o
/home/travis/build/boostorg/compute/test/test_scatter_if.cpp:38:21: error: 
      non-aggregate type 'bc::vector<int>' cannot be initialized with an
      initializer list
    bc::vector<int> output = {-1, -1, -1, -1, -1, -1, -1, -1, -1, -1};
                    ^        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

However, you should be able to change this to use the size and constant constructor like so:
```
bc::vector<int> output(input.size, -1, queue);
```

**Comment by kylelutz** [comment]
Merged! Thanks!

**Pull request merged**